docs/guides/c-extension-for-python.mdx

```diff
@@ -0,0 +1,230 @@
+
+# now we can directly call the function
+obs = lib.build_observable()
+print("SparseObservable instance?", isinstance(obs, qiskit.quantum_info.SparseObservable))
+print(obs)
+```
+
+## Build
+
+First, you have to build the Qiskit Python extension. This includes the C symbols so you can access both interfaces via the same shared library. This is important to ensure data can be passed
+correctly across C and Python.
+```bash
+python setup.py build_rust --inplace --release
+```
+The shared library is called ``_accelerate.<platform-specific-part>``, whose location and name we can find as
+```bash
+QKLIB=$(python -c "import os; import qiskit; print(os.path.dirname(qiskit._accelerate.__file__))")
+QKNAME=$(python -c "import os; import qiskit; print(os.path.basename(qiskit._accelerate.__file__))")
+```
+
+We will need to know the location of the environments Python includes (``Python.h``) and libraries (``libpython.<suffix>``).
+These can, for example, be identified via
+```bash
+PYINCLUDE=$(python -c "import sysconfig; print(sysconfig.get_path('include'))")
+PYLIB=$(python -c "import sysconfig; print(sysconfig.get_config_var('LIBDIR'))")
+PYNAME=$(find $PYLIB -maxdepth 1 -name "libpython*" | grep -oE "[^/]+$" | grep -oE "python[0-9]+\.[0-9]+" || echo "python")
+```
+(If you just know these locations and names, you can also just set them directly.)
+
+### Linux
+
+Linking can slightly differ between platforms, here we first describe a solution for linkers
+supporting libraries with arbitrary names, using the `-l:` flag. See below if your linker
+requires the library to be called `lib<something>`.
+
+We can build the extension specifying the full name of the `_accelerate` library:
+```bash
+gcc extension.c -fpic -shared -o cextension.so \
+  -I/path/to/dist/c/include -L$QKLIB -l:$QKNAME \
+  -I$PYINCLUDE -L$PYLIB -l$PYNAME
+```
+And then simply ``python main.py`` to run the Python program.
+
+### MacOS
+
+An alternative to using the exact library name with `-l:` is to symlink the `_accelerate` library
+to the desired name. This solution works on Mac (and also Linux if you replace all occurences of `.dylib` with `.so`).
+To include the `_accelerate` shared library we symlink it to the linkers expected format of `lib<library name>.<suffix>`,
+```bash
+ln -s $QKLIB/$QKNAME $QKLIB/libqiskit.dylib
+```
+which allows to use `qiskit` as library name:
+```bash
+gcc extension.c -fpic -shared -o qiskit_cextension.so \
+  -I/path/to/dist/c/include -L$QKLIB -lqiskit \
+  -I$PYINCLUDE -L$PYLIB -l$PYNAME
+```
+Then simply ``python main.py`` to run the Python program.
+
+## Manual C extension
+
+Instead of using `ctypes`, it is possible to manually build an extension for Python using [Python's
+C API](https://docs.python.org/3/c-api/intro.html) directly. This has the potential to be
+faster than using `ctypes`, albeit requires more effort to implement.
+In the following we provide a brief example on how this can be achieved
```

Comment:
```suggestion
Instead of using `ctypes`, it is possible to manually build an extension for Python using [Python's
C API](https://docs.python.org/3/c-api/intro.html) directly. This has the potential to be
faster than using `ctypes`, though it requires more effort to implement.
The following code is a brief example on how to achieve this.
```